lib/Matrix.js

```diff
@@ -138,7 +142,20 @@ function useMatrixProvider(auth) {
             }
         }
         if ((membership === 'invite')) {
-            setInvites(setRoom(room.roomId));
+            // we need to update the matrixClient room sync so getRooms(roomId) in buildRoomObject doesn't return null
+            await matrixClient.roomInitialSync(room.roomId)
```

Comment:
Find it hard to explain this quickly, but as a rule of thumb: Never do any `async` actions inside the event handler functions.

[These functions](https://github.com/medienhaus/medienhaus-spaces/pull/49/files#diff-2f8d4752f5ecbf336e6d4aeb24e15c2483c854d3d0c24b137625b5ad3376d301L186-L188) get executed for each and every single event of those types that can be found in the Matrix. This includes **all events** during the initial sync at the beginning, and this includes every event of those types that happens afterwards when the user has the page open.

To get around this let's think about another way where we can do this: If I understood correctly you need this in order to display detailed information per invitation on the dashboard.
Do we need that detailed information at all times? Or do we only need it in the dashboard?

If it's the latter, could you think of a way of fetching that `inviter` information only when looking at the dashboard?

If this gets too complicated I'm happy to make these changes myself, after you've addressed the other review comments.

Reply:
changed it to fetch the information within dashboard.

If we want to display invitations for individual services on the actual service page, as well as on the dashboard it will probably make sense to fetch this inforamtion directly when we receive an invitation. But then let's cross that bridge when we get to it 🙂